### DISCUSSION TITLE
Vehicle Routing with Pickups and Deliveries with no connections between two points

### DISCUSSION THREAD

**Discussion by lutz500** [google/or-tools#3285] [Routing (and legacy CP) questions]
The Vehicle Routing with Pickups and Deliveries instruction is based on the distance matrix. In all examples there is always a connection from each Location the all other locations. 
Is it possible to enter a certaine value like -1 to show that there is no connection between to locations?

**Comment by Mizux** [comment]
instead of `-1` you should instead use a number larger than the vehicle max capacity value you set upon creating RoutingDimension.
Solver don't allow "CumulVar overflow"
e.g. 

```py
# Create and register a transit callback.
def distance_callback(from_index, to_index):
    """Returns the distance between the two nodes."""
    # Convert from routing variable Index to distance matrix NodeIndex.
    from_node = manager.IndexToNode(from_index)
    to_node = manager.IndexToNode(to_index)
    return data['distance_matrix'][from_node][to_node]

transit_callback_index = routing.RegisterTransitCallback(distance_callback)

# Define cost of each arc.
routing.SetArcCostEvaluatorOfAllVehicles(transit_callback_index)

# Create a Distance Routing Dimension
dimension_name = 'Distance'
routing.AddDimension(
    transit_callback_index,
    0,  # no slack
    42_000,  # HERE vehicle maximum travel distance
    True,  # start cumul to zero
    dimension_name)
```
here, if in my distance matrix I return a number higher than `42_000`  it will effectively forbid any vehicle to take this arc.
